nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -298,11 +324,15 @@ private void acquireAccessDetails() {
         }
 
         RequestBody acquireTokenRequestBody = acquireTokenBuilder.build();
-
-        Request acquireTokenRequest = new Request.Builder()
+        Request.Builder acquireTokenRequestBuilder = new Request.Builder()
             .url(authorizationServerUrl)
-            .post(acquireTokenRequestBody)
-            .build();
+            .post(acquireTokenRequestBody);
+
+        if (clientAuthenticationStrategy.equals(BASIC_AUTHENTICATION.getValue()) && clientId != null) {
+            acquireTokenRequestBuilder.addHeader("Authorization", Credentials.basic(clientId, clientSecret));
```

Comment:
Recommend defining `private static final String AUTHORIZATION_HEADER = "Authorization"` and reusing that value here and in the `refreshAccessDetails()` method.